src/main/java/com/firebolt/jdbc/type/array/SqlArrayUtil.java

```diff
@@ -221,4 +225,31 @@ private static Object[] toObjectArray(Object array) {
 		}
 		return ret;
 	}
+
+	public static String byteArrayToHexString(byte[] bytes, boolean separateEachByte) {
+		if (bytes == null)  {
+			return null;
+		}
+		ByteBuffer buffer = ByteBuffer.wrap(bytes);
+		String separator = separateEachByte ? BYTE_ARRAY_PREFIX : "";
+		return BYTE_ARRAY_PREFIX + Stream.generate(buffer::get).limit(buffer.capacity()).map(Integer::toHexString).collect(joining(separator));
+	}
+
+	@SuppressWarnings("java:S1168") // we have to return null here
+	public static byte[] hexStringToByteArray(String str) {
+		if (str == null) {
+			return null;
+		}
+		if (!str.startsWith(BYTE_ARRAY_PREFIX)) {
+			return str.getBytes();
+		}
+		char[] chars = str.substring(2).toCharArray();
+		byte[] bytes = new byte[chars.length / 2];
+		for (int i = 0; i < chars.length; i += 2) {
+			bytes[i / 2] = (byte) ((Character.digit(chars[i], 16) << 4) + Character.digit(chars[i + 1], 16));
+		}
```

Comment:
Is there a way to use `getBytes` here as well? Perhaps stripping `BYTE_ARRAY_PREFIX` from every place and then calling it? This way we don't have to maintain this bitwise shifting logic.